components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/security/CryptoUtil.java

```diff
@@ -0,0 +1,122 @@
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.micro.integrator.initializer.handler.transaction.security;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.olingo.commons.api.Constants;
+import org.wso2.micro.integrator.initializer.handler.transaction.TransactionConstants;
+import org.wso2.micro.integrator.initializer.handler.transaction.TransactionException;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * This class will provide the required methods to encrypt a given value.
+ */
+public class CryptoUtil {
+
+    private static final Log LOG = LogFactory.getLog(CryptoUtil.class);
+
+    private CryptoUtil() {
+    }
+
+    /**
+     * Initializes the Cipher.
+     *
+     * @return cipher.
+     * @throws TransactionException - when something goes wrong while initializing the cipher.
+     */
+    public static Cipher initializeCipher() throws TransactionException {
+        Cipher cipher;
+        KeyStore primaryKeyStore = getKeyStore(getAbsolutePathToKeyStoreLocation(),
+                                               TransactionConstants.KEYSTORE_PASSWORD,
+                                               TransactionConstants.KEY_TYPE);
+        try {
+            Certificate certs = primaryKeyStore.getCertificate(TransactionConstants.KEY_ALIAS);
+            cipher = Cipher.getInstance(TransactionConstants.ENCRYPTION_ALGORITHM);
+            cipher.init(Cipher.ENCRYPT_MODE, certs);
+            LOG.debug("Cipher for Transaction Handler component is initialized Successfully");
+        } catch (KeyStoreException | NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException e) {
+            throw new TransactionException("Error initializing Cipher ", e);
```

Comment:
Cipher for what? EI can use ciphers for many different things.

Reply:
+1